lib/matplotlib/tests/test_image.py

```diff
@@ -1218,3 +1219,69 @@ def test_huge_range_log(fig_test, fig_ref):
     ax = fig_ref.subplots()
     im = ax.imshow(data, norm=colors.Normalize(vmin=100, vmax=data.max()),
                    interpolation='nearest', cmap=cmap)
+
+
+@check_figures_equal()
+def test_spy_box(fig_test, fig_ref):
+    # setting up reference and test
+    ax_test = fig_test.subplots(1, 3)
+    ax_ref = fig_ref.subplots(1, 3)
+
+    # plotting with spy
+    ax_test[0].set_title("ones")
+    ax_test[0].spy([[1, 1], [1, 1], ])
+    ax_test[1].set_title("zeros")
+    ax_test[1].spy([[0, 0], [0, 0], ])
+    ax_test[2].set_title("mixed")
+    ax_test[2].spy([[0, 1], [1, 0], ])
+
+    # plotting with imshow
```

Comment:
You have the same code repeated three times here.  Suggest you rewrite as a loop? i.e. 

```
for z, title in zip([[1, 1], [1, 1]], [[0, 0], [0, 0]], [[0, 1], [1, 0]], ['ones', 'zeros', 'both']):
```